src/Control/Effect.hs

```diff
@@ -48,114 +54,50 @@ class Effect sig where
   -- | Higher-order functor map of a natural transformation over higher-order positions within the effect.
   hfmap :: (forall x . m x -> n x) -> sig m a -> sig n a
 
-  -- | Handle any effects in higher-order positions by threading the 'Carrier'’s state all the way through to the continuation.
-  --
-  --   For first-order effects (which don’t contain higher-order positions), this will simply involve repackaging the effect’s arguments at the new type.
-  --
-  --   For higher-order effects (which do contain higher-order positions), the 'Carrier' actions @c n@ must be 'resume'd to obtain the necessary @n@ action, and the state passed along to the continuation via 'resume' and 'wrap'.
-  handle :: (Carrier c f, Monad n)
+  -- | Handle any effects in higher-order positions by threading the carrier’s state all the way through to the continuation.
+  handle :: (Functor f, Monad n)
          => f ()
-         -> sig (c n) (c n a)
-         -> sig n (c n a)
+         -> (forall x . f (m x) -> n (f x))
+         -> sig m (m a)
+         -> sig n (n (f a))
+
+
+class Effect sig => TermAlgebra h sig | h -> sig where
+  var :: a -> h a
+  con :: sig h (h a) -> h a
```

Comment:
`var` and `con` must be used to construct effectful computations for correctness’ sake. Fortunately this is extremely easy to abide by since `send` &c. use these.